### ISSUE TITLE
Add support for async transformRequest

### ISSUE THREAD

**Issue by PunKHS** [appbaseio/reactivesearch#1114]
**Affected Projects**
React

Hello, I have a feature request to ReactiveBase component. It would be great if you could add support for async transformRequest functions.

The goal I want to achieve is to add JWT token to the authorization header, but this token must be fetched from external service, and sometimes must be refreshed, so the function which manages it is asynchronous.

It'd be great if that code could work:
```
<ReactiveBase
    transformRequest={
        async request => ({
            ...request,
            headers: {
                ...request.headers,
                Authorization: `Bearer ${await AuthService.getToken()}`
            },
        })
    }
>
</ReactiveBase>
```
**Library Version:** 3.0.0-rc.26

**Comment by PunKHS** [comment]
May be somebody knows a workaround, which can help resolve this issue?

**Comment by bietkul** [comment]
Fixed in v3.0.1.
